trustchain-ion/src/attestor.rs

```diff
@@ -268,6 +309,49 @@ mod tests {
         assert!(vc_with_proof.is_ok());
     }
 
+    #[tokio::test]
+    async fn test_attest_presentation() {
+        // Note: removed tmp directory overwrite for TRUSTCHAIN_DATA, to have access to
+        // the signing keys in .trustchain
+        // init();
+        let resolver = get_ion_resolver("http://localhost:3000/");
+        // root+1
+        let issuer_did = "did:ion:test:EiBVpjUxXeSRJpvj2TewlX9zNF3GKMCKWwGmKBZqF6pk_A";
+        // root+2
+        let holder_did = "did:ion:test:EiAtHHKFJWAk5AsM3tgCut3OiBY4ekHTf66AAjoysXL65Q";
+
+        let issuer = IONAttestor::new(issuer_did);
+        let holder = IONAttestor::new(holder_did);
+        // let target = IONAttestor::try_from(AttestorData::new(
+        //     did.to_string(),
+        //     serde_json::from_str(TEST_SIGNING_KEYS).unwrap(),
+        // ))
+        // .unwrap();
+        let vc = serde_json::from_str(TEST_CREDENTIAL).unwrap();
+        let vc_with_proof = issuer.sign(&vc, None, None, &resolver).await.unwrap();
+        let presentation = Presentation {
+            verifiable_credential: Some(OneOrMany::One(CredentialOrJWT::Credential(vc_with_proof))),
+            ..Default::default()
+        };
+        // assert holder field is not initially populated
+        assert!(presentation.holder.is_none());
+
+        // Attest to vp:
+        // .sign_presenatation now checks and sets the holder field of the presentation
+        // This has implications for the proof generation which is handled by the ssi library:
+        //   - ssi::ldp::ensure_or_pick_verification_relationship calls presentation.get_issuer()
+        //      which returns the holder (if Some)
+        //   - ensure_or_pick_verification_relationship tries to resolve the did and check it's
+        //      verification methods
```

Comment:
Move to above `sign_presentation` method in the `Holder` trait

Reply:
I've moved it to the implementation of the trait in tc-ion, where we introduce the principal of enforcing that the holder's field is populated